src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -83,12 +83,12 @@ public class GoPiGo : IDisposable
         /// <param name="spiDevice">The SpiDevice</param>
         /// <param name="spiAddress">The SPI address, by default 8</param>
         /// <param name="autoDetect">Try to autodetect the board</param>
-        /// <param name="autoDispose">True to dispose the SpiDevice when disposing the class</param>
-        public GoPiGo(SpiDevice spiDevice, byte spiAddress = 8, bool autoDetect = true, bool autoDispose = true)
+        /// <param name="shouldDispose">True to dispose the SpiDevice when disposing the class</param>
+        public GoPiGo(SpiDevice spiDevice, byte spiAddress = 8, bool autoDetect = true, bool shouldDispose = true)
         {
-            _spiDevice = spiDevice ?? throw new ArgumentException("SpiDevice can't be null");
+            _spiDevice = spiDevice ?? throw new ArgumentException(nameof(spiDevice));
```

Comment:
Should be `ArgumentNullException` here.

Reply:
You can make suggestions, like this.
```suggestion
            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
```

Reply:
Oh, cool, will do next 👍